src/libstore/daemon.cc

```diff
@@ -1032,6 +1032,15 @@ void processConnection(
     if (GET_PROTOCOL_MINOR(clientVersion) >= 33)
         to << nixVersion;
 
+    if (GET_PROTOCOL_MINOR(clientVersion) >= 35) {
+        // We and the underlying store both need to trust the client for
+        // it to be trusted.
+        auto temp = trusted
+            ? store->isTrustedClient()
+            : std::optional { NotTrusted };
+        to << (temp ? 1 + (uint8_t)*temp : (uint8_t)0);
```

Comment:
:exploding_head: 

Can we rather have a proper well-defined serialization scheme for `std::optional<TrustedFlag>` (in `worker-protocol.hh` like the rest I would assume)? This inlined arithmetic trick is too smart for its own good I think